database_apis/users_database.py

```diff
@@ -117,12 +119,44 @@ def logout(self):
         
     def get_current_user(self):
         return self.user
+    
+    def add_auto_login_information(self, mac_address : str, email: str):
+        self.db.execute( "UPDATE users SET mac_address = ? WHERE email = ?", (mac_address, email,) )
+        self.db.execute( "UPDATE users SET date = datetime('now', '+7 day') WHERE email = ?", (email,) )
+        self.connection.commit()
+
+    def auto_login(self, mac_address : str):                #problem here
```

Comment:
This function uses the user's mac address to retrieve user's email and password to login. Since the email and password is taken from the user table, it would work even if the user changed their email and password while auto login is effective.

Reply:
Note that I didn't put add_auto_login_information with functions that changes user's data since auto login information is updated more frequently.